pyproject.toml

```diff
@@ -64,6 +64,8 @@ dev = [
     "setuptools_scm",
     # test speedups
     "pytest-xdist",
+    # until https://github.com/twisted/towncrier/pull/651 is released
+    "towncrier @ git+https://github.com/twisted/towncrier",
```

Comment:
you could move this to `hatch.toml`, as having these URLs directly in `project.(optional-)dependencies` technically goes against spec.

that way we also don’t need to add `tool.hatch.metadata.allow-direct-references = true`